File: examples/CBuildr6/ExMruU.cpp
//---------------------------------------------------------------------------
#include <vcl\vcl.h>
#pragma hdrstop

#include "ExMruU.h"
//---------------------------------------------------------------------------
#pragma link "OvcState"
#pragma link "OvcBase"
#pragma link "OvcStore"
#pragma link "OvcFiler"
#pragma link "OvcMRU"
#pragma resource "*.dfm"
TForm1 *Form1;
//---------------------------------------------------------------------------
__fastcall TForm1::TForm1(TComponent* Owner)
	: TForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TForm1::Open1Click(TObject *Sender)
{
  // Show the file open dialog.
  if (OpenDialog1->Execute())  {
    // Load the file into the Memo.
    Memo1->Lines->LoadFromFile(OpenDialog1->FileName);
    // Add all files to the main MRU list.
    FileMenuList->Add(OpenDialog1->FileName);
    // If a C++Builder BPR file was opened, add it to the top
    // of the split MRU list.
    if (UpperCase(ExtractFileExt(OpenDialog1->FileName)) == ".BPR" ||
        UpperCase(ExtractFileExt(OpenDialog1->FileName)) == ".MAK")  {
      BCBList->AddSplit(OpenDialog1->FileName, apTop);
      Reopen1->Enabled = true;
    }
    // If a C++Builder CPP file was opened, add it to the bottom
    // of the split MRU list.
    if (UpperCase(ExtractFileExt(OpenDialog1->FileName)) == ".CPP")  {
      BCBList->AddSplit(OpenDialog1->FileName, apBottom);
      Reopen1->Enabled = true;
    }
  }
}
//---------------------------------------------------------------------------
void __fastcall TForm1::ClearHistoryList1Click(TObject *Sender)
{
  BCBList->Clear();
  FileMenuList->Clear();
  Reopen1->Enabled = false;
}
//---------------------------------------------------------------------------
void __fastcall TForm1::Exit1Click(TObject *Sender)
{
  Close();
}
//---------------------------------------------------------------------------
void __fastcall TForm1::MenuMRUClick(TObject *Sender,
      const AnsiString ItemText, TOvcMRUClickAction &Action)
{
  // A MRU menu item was clicked. Open the file in the Memo.
  Memo1->Lines->LoadFromFile(ItemText);
}
//---------------------------------------------------------------------------

File: examples/Delphi/EXMRUU.pas
{$J+} {Writable constants}

unit ExMruU;

interface

uses
  {$IFDEF WIN32}
  Windows,
  {$ELSE}
  WinTypes, WinProcs,
  {$ENDIF}
  Messages, SysUtils, Classes, Graphics, Controls, Forms, Dialogs,
  Menus, OvcBase, OvcMRU, StdCtrls, OvcState, OvcFiler, OvcStore;

type
  TForm1 = class(TForm)
    MainMenu1: TMainMenu;
    File1: TMenuItem;
    Exit1: TMenuItem;
    N1: TMenuItem;
    Open1: TMenuItem;
    OpenDialog1: TOpenDialog;
    Memo1: TMemo;
    Reopen1: TMenuItem;
    ClearHistoryList1: TMenuItem;
    OvcComponentState1: TOvcComponentState;
    OvcIniFileStore1: TOvcIniFileStore;
    DelphiList: TOvcMenuMRU;
    FileMenuList: TOvcMenuMRU;
    procedure Open1Click(Sender: TObject);
    procedure ClearHistoryList1Click(Sender: TObject);
    procedure Exit1Click(Sender: TObject);
    procedure OvcMRUClick(Sender: TObject; const ItemText: String;
      var Action: TOvcMRUClickAction);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  Form1: TForm1;

implementation

{$R *.DFM}

procedure TForm1.Open1Click(Sender: TObject);
begin
  { Show the file open dialog. }
  if (OpenDialog1.Execute) then begin
    { Load the file into the Memo. }
    Memo1.Lines.LoadFromFile(OpenDialog1.FileName);
    { Add all files to the main MRU list. }
    FileMenuList.Add(OpenDialog1.FileName);
    { If a Delphi DPR file was opened, add it to the top }
    { of the split MRU list. }
    if UpperCase(ExtractFileExt(OpenDialog1.FileName)) = '.DPR' then begin
      DelphiList.AddSplit(OpenDialog1.FileName, apTop);
      Reopen1.Enabled := True;
    end;
    { If a Delphi PAS file was opened, add it to the bottom }
    { of the split MRU list. }
    if UpperCase(ExtractFileExt(OpenDialog1.FileName)) = '.PAS' then begin
      DelphiList.AddSplit(OpenDialog1.FileName, apBottom);
      Reopen1.Enabled := True;
    end;
  end;
end;

procedure TForm1.ClearHistoryList1Click(Sender: TObject);
begin
  DelphiList.Clear;
  FileMenuList.Clear;
  Reopen1.Enabled := False;
end;

procedure TForm1.Exit1Click(Sender: TObject);
begin
  Close;
end;

procedure TForm1.OvcMRUClick(Sender: TObject; const ItemText: String;
  var Action: TOvcMRUClickAction);
begin
  { A MRU menu item was clicked. Open the file in the Memo. }
  Memo1.Lines.LoadFromFile(ItemText);
end;

end.
